contracts/lbp/LBPManagerFactory.sol

```diff
@@ -0,0 +1,141 @@
+/*
+
+██████╗░██████╗░██╗███╗░░░███╗███████╗██████╗░░█████╗░░█████╗░
+██╔══██╗██╔══██╗██║████╗░████║██╔════╝██╔══██╗██╔══██╗██╔══██╗
+██████╔╝██████╔╝██║██╔████╔██║█████╗░░██║░░██║███████║██║░░██║
+██╔═══╝░██╔══██╗██║██║╚██╔╝██║██╔══╝░░██║░░██║██╔══██║██║░░██║
+██║░░░░░██║░░██║██║██║░╚═╝░██║███████╗██████╔╝██║░░██║╚█████╔╝
+╚═╝░░░░░╚═╝░░╚═╝╚═╝╚═╝░░░░░╚═╝╚══════╝╚═════╝░╚═╝░░╚═╝░╚════╝░
+
+*/
+
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// solium-disable linebreak-style
+pragma solidity ^0.8.6;
+
+import "../utils/CloneFactory.sol";
+import "openzeppelin-solidity/contracts/access/Ownable.sol";
+import "./LBPManager.sol";
+
+/**
+ * @title PrimeDAO LBPManager Factory
+ * @dev   Enable PrimeDAO governance to create new LBPManager contracts.
+ */
+contract LBPManagerFactory is CloneFactory, Ownable {
+    address public lbpManagerMasterCopy;
+    address public LBPFactory;
+
+    event LBPDeployedUsingManager(
+        address indexed lbp,
+        address indexed lbpManager,
+        address indexed admin
+    );
+
+    /**
+     * @dev                             Constructor.
+     * @param _LBPFactory               The address of Balancers LBP factory.
+     */
+    constructor(address _LBPFactory) {
+        require(
+            _LBPFactory != address(0),
+            "LBPManagerFactory: LBPFactory can not be zero"
+        );
+        LBPFactory = _LBPFactory;
+    }
+
+    /**
+     * @dev                             Set LBPManager contract which works as a base for clones.
+     * @param _masterCopy               The address of the new LBPManager basis.
+     */
+    function setMasterCopy(address _masterCopy) external onlyOwner {
+        require(
+            _masterCopy != address(0),
+            "LBPManagerFactory: mastercopy can not be zero"
+        );
+        require(
+            _masterCopy != address(this),
+            "LBPManagerFactory: mastercopy can not be the same as LBPManagerFactory"
+        );
+        lbpManagerMasterCopy = _masterCopy;
+    }
+
+    /**
+     * @dev                             Set Balancers LBP Factory contract as basis for deploying LBPs.
+     * @param _LBPFactory               The address of Balancers LBP factory.
+     */
+    function setLBPFactory(address _LBPFactory) external onlyOwner {
+        require(
```

Comment:
logic of require statements on line 53, 57, 68 and 72 is the same, maybe it makes sense to create a generic modifier for it